sol/contracts/ics23.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.2;
+import {BatchProof, CompressedBatchProof, CommitmentProof, ProofSpec, ExistenceProof, NonExistenceProof} from "./proofs.sol";
+import {Compress} from "./ics23Compress.sol";
+import {Proof} from "./ics23Proof.sol";
+import {Ops} from "./ics23Ops.sol";
+import {BytesLib} from "GNSPS/solidity-bytes-utils@0.8.0/contracts/BytesLib.sol";
+
+library Ics23  {
+
+    // verifyMembership, throws an exception in case anything goes wrong
+    function verifyMembership(ProofSpec.Data memory spec, bytes memory commitmentRoot, CommitmentProof.Data memory proof, bytes memory key, bytes memory value) internal pure {
```

Comment:
the function should return bool. If the proof is invalid, that's okay, let the user handle it. `require` will bail out and revert the execution

Reply:
I see I see :thinking: 
yeah it's a legit request 
I can refactor all the functions to return a bool as second element in sort of a Go error handling style...
but then we would lose the information about the error :thinking: 
Maybe I can write an enum to represent the error, there would be a lot of enums, but at least we wouldn't lose information about the error...

Reply:
Hi @mkaczanowski ,
my last commit changes the return values of functions.
Function callers will get an enum encoding the Error, no more require and revert except those in the auto-generated bindings